osu.Server.Spectator/Hubs/SpectatorHub.cs

```diff
@@ -119,6 +126,8 @@ public async Task EndPlaySession(SpectatorState state)
                     score.ScoreInfo.Date = DateTimeOffset.UtcNow;
 
                     scoreUploader.Enqueue(scoreToken.Value, score);
+                    await scoreProcessedSubscriber.RegisterForNotificationAsync(Context.ConnectionId, CurrentContextUserId, scoreToken.Value,
+                        (connectionId, userId, scoreId) => Clients.Client(connectionId).UserScoreProcessed(userId, scoreId));
```

Comment:
This callback has a closure to the hub instance. I'm not sure if I'm allowed to do this, but I'm not sure how *else* I'm supposed to be delivering the async message otherwise. I guess I could have some sort of memory-based polling wherein the hub queries the subscriber if it has anything new to notify about? But can you even have a recurring timer in a hub? No idea.

Do note though that I am attempting to be extra careful to capture all of the ambient state of the hub (`Context.ConnectionId`, `CurrentContextUserId`) at the point of call. Not doing so would risk misfiring the message to a completely random user at some point in the future.

Reply:
Never mind I am dumb and didn't think hard enough what to google. Googling "hub lifetime" spells out [pretty clearly](https://github.com/dotnet/AspNetCore.Docs/issues/6888) that I was doing Bad Things. Thankfully there's also https://learn.microsoft.com/en-us/aspnet/core/signalr/hubcontext?view=aspnetcore-2.1 and it works and I fixed it.